content/docs/tasks/external_auth_opa.md

```diff
@@ -0,0 +1,133 @@
+
+OSM allows configuring envoy's [External Authorization extension](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/ext_authz_filter) through OSM's MeshConfig.
+Authorization filter is currently applied to inmesh `inbound` and `ingress` connections affecting HTTP request made towards any service in the mesh.
+
+The following example will show how to configure external authorization with `opa-envoy-plugin`.
+Note this example will use a single, remote (over the network) endpoint to validate all traffic. This is not a suggested production deployment.
+
+Example OSM's MeshConfig:
+
+```
+...
+inboundExternalAuthorization:
+  enable: true
+  address: opa.opa.svc.cluster.local
+  port: 9191
+  statPrefix: authz_opa
+  timeout: 1s
+  failureModeAllow: false
+```
+
+## Example Walkthrough
+
+- Deploy an [opa-envoy-plugin](https://github.com/open-policy-agent/opa-envoy-plugin), you will need to back the service with network connectivity for this walkthrough - we highly suggest to read carefully `ola-envoy-plugin`'s guide first.
+```
+kubectl apply -f https://raw.githubusercontent.com/open-policy-agent/opa-envoy-plugin/main/quick_start.yaml
+```
+Once `opa-envoy-plugin` is running and reachable through the network, you can proceed to deploy OSM's Demo, follow `demo/run-osm-demo.sh`
+```
+demo/run-osm-demo.sh  # wait for all services to come up
+```
+
+Followingly, edit OSM's meshconfig to add external authorization to the mesh pods:
+```
+kubectl edit meshconfig osm-mesh-config -n osm-system
+...
+inboundExternalAuthorization:
+      enable: true
+      address: opa.opa.svc.cluster.local  ## or reachable FQDN for opa plugin running
+      port: 9191
+      failureModeAllow: false
+      statPrefix: inboundExtAuthz
+      timeout: 1s
+
+```
+
+Traffic should fail right out of the bat:
+```
+kubectl logs <bookbuyer_pod> -n bookbuyer bookbuyer
+```
+```
+...
+--- bookbuyer:[ 8 ] -----------------------------------------
+
+Fetching http://bookstore.bookstore:14001/books-bought
+Request Headers: map[Client-App:[bookbuyer] User-Agent:[Go-http-client/1.1]]
+Identity: n/a
+Booksbought: n/a
+Server: envoy
+Date: Tue, 04 May 2021 01:20:39 GMT
+Status: 403 Forbidden
+ERROR: response code for "http://bookstore.bookstore:14001/books-bought" is 403;  expected 200
+...
+```
+
+You shoud also be able to see the logs in `opa-envoy-plugin` for rejected authorization calls:
+```
+kubectl logs opa -n opa
+...
+{"decision_id":"1df154b5-658a-47bf-ac18-be52998605da"
+...
+"result":false,   // resulting decision
+...
+"time":"2021-05-04T01:21:18Z","timestamp":"2021-05-04T01:21:18.1971808Z","type":"openpolicyagent.org/decision_logs"
+}
+```
+
+- Now edit OPA's policy:
```

Comment:
```suggestion
## Update the OPA policy

Edit the OPA policy using `kubectl edit`:
- Now edit OPA's policy:
```

Reply:
let's discuss after addressing latest comments how to do this.